src/incremental/updateCil.ml

```diff
@@ -89,8 +90,9 @@ let update_ids (old_file: file) (ids: max_ids) (new_file: file) (changes: change
       List.iter (assign_same_id f.sallstmts) d.unchangedNodes
   in
   let reset_changed_globals (changed: changed_global) =
-    match (changed.current, changed.old) with
-    | GFun (nw, _), GFun (old, _) -> reset_changed_fun nw old changed.unchangedHeader changed.diff
+    match (changed.current.def, changed.old.def) with
+    | Some (Fun nw), Some (Fun old) -> reset_changed_fun nw old changed.unchangedHeader changed.diff
+    (* why ids of global variables not adapted? *)
```

Comment:
We should probably figure this out here. Also regarding just declarations.

Reply:
I discussed with @jerhard  that it is wrong to not change the id of changed globals since then the uniqueness of the ids is not necessarily preserved. We think a change to the global variable will often lead to a lot of re-evaluations anyway and adapting the old ids could lead to a lot of precision loss. Therefore, we decided to generate new ids for changed globals.

Reply:
>Also, we were thinking that we should maybe generate new ids for functions with changed headers also

That should be possible. Did you come to a  conclusion?

Reply:
There are a couple of incremental tests failing after implementing this. One reason seems to be that functions included in `exclude_from_rel_destab` (as for example force-reanalyze functions) are destabilized at their entry node to force reevaluation. If the id of these functions (which are assumed to be completely changed and not have an unchanged header) changes, the destabilization can not go through all the old influences of this entry node and therefore does not destabilize enough to actually trigger reevaluation.